code-challenges/for-each/challenges-01.test.js

```diff
@@ -0,0 +1,200 @@
+  }
+
+  return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named removeOne that takes in a number and an array. If the number divided by three has a remainder of two, pop one element off of the array.
+
+Hint: you may want to look into the modulo operation.
+
+Then, write a function named removeElements that takes in an array and a callback. This function should use a for loop to iterate over the array and invoke the callback once for each element in the array.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const removeOne = (num, arr) => {
+  if(num % 3 === 2) {
+    arr.pop(num);
+  }
+}
+
+const removeElements = (arr, callback) => {
+  for(let i = 0; i < arr.length; i++){
+    callback(arr[i], arr)
+  }
+  return arr;
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named removeWithForEach that produces the same output as challenge 3, but uses forEach.
+------------------------------------------------------------------------------------------------ */
+
+const removeWithForEach = (arr, callback) => {
+  arr.forEach(function(currentValue) {
+    callback(currentValue, arr)
+  })
+  return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named removeWithAnon that produces the same output as challenges 3 and 4.
+
+This function should use forEach again, but rather than taking in a callback as an argument, define an anonymous function as the argument to forEach.
+
+This anonymous function should accept up to three arguments: the element, the index, and the array.
+------------------------------------------------------------------------------------------------ */
+
+const removeWithAnon = (arr) => {
+  arr.forEach(function(currentValue) {
+    removeOne(currentValue, arr)
+  })
+  return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+Write a function named createList that takes in an array of the current store intentory.
+
+The inventory is formatted like this:
+[
+  { name: 'apples', available: true },
+  { name: 'pears', available: true },
+  { name: 'oranges', available: false },
+  { name: 'bananas', available: true },
+  { name: 'blueberries', available: false }
+]
+
+This function should use forEach to populate your grocery list based on the store's inventory. If the item is available, add it to your list. Return the final list.
+------------------------------------------------------------------------------------------------ */
+
+const createList = (availableItems) => {
+  let storeInventory = []
+  availableItems.forEach(function(item){
+    if(item == true) storeInventory.push(item)

```

Comment:
You should see a linter error here once you have your linter working properly. Fix the suggested issue.